### PR TITLE
Yearly goal estimate doesn't work

### PR DESCRIPTION
**Pull request by Troff8**
## PR includes

- [x] Bug Fix

## Related issues

Resolve [#1292 ](https://github.com/taskany-inc/issues/issues/1292)

### CODE REVIEW

File: src/utils/db.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ export const findOrCreateEstimate = async (
 
     if (estimate.id == null) {
         const { id: _, ...whereParams } = currentEstimate;
```
Comment (9teen90nine) [inline src/utils/db.ts:23]: `const { id: _, q = null, date = null} = currentEstimate;`

`findFirst({where: { q, date, ...whereParams })`

File: src/utils/db.ts [hunk 2]
```diff
@@ -22,9 +22,10 @@ export const findOrCreateEstimate = async (
     let currentEstimate = { ...estimate };
 
     if (estimate.id == null) {
-        const { id: _, ...whereParams } = currentEstimate;
+        const { id: _, q = null, date = null, ...whereParams } = currentEstimate;
```
Comment (awinogradov) [inline src/utils/db.ts:25]: Why this is not `estimate`? Do we really need a copy of it?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/utils/db.ts (hunk 2 above)
Comment (Troff8) [reply src/utils/db.ts]: like a copy is not needed, now made without a copy

File: src/utils/db.ts [hunk 3]
```diff
@@ -19,18 +19,17 @@ export const findOrCreateEstimate = async (
...
-        const { id: _, ...whereParams } = currentEstimate;
+        const { id: _, q = null, date = null, ...whereParams } = estimate;
+
         const realEstimate = await prisma.estimate.findFirst({
-            where: whereParams,
+            where: { q, date, ...whereParams },
         });
 
         if (realEstimate) {
-            currentEstimate.id = realEstimate.id;
```
Comment (awinogradov) [inline src/utils/db.ts:31]: Replace all this code with code below

``` ts
return existingEstimate || prisma.estimate.create({
                data: {
                    date: estimate.date,
                    y: estimate.y,
                    q: estimate.q,
                    activityId,
                    goalId,
                },
            });
```

**Review by awinogradov — CHANGES_REQUESTED** [review]
You can do it better

**Review by awinogradov — APPROVED** [review]

**Pull request merged**